src/scratchOrgSettingsGenerator.ts

```diff
@@ -134,9 +135,16 @@ export default class SettingsGenerator {
           logger.debug(`An error occurred trying to check deploy id: ${id}`);
           logger.debug(`Error: ${(error as Error).message}`);
           logger.debug('Re-trying deploy check again....');
-          return {
-            completed: false,
-          };
+          if (
+            ['ETIMEDOUT', 'ENOTFOUND', 'ECONNRESET', 'socket hang up'].some((retryableNetworkError) =>
+              (error as Error).message.includes(retryableNetworkError)
```

Comment:
we're casting `error as Error` three times here, should we just set it as a `const err = error as Error`?